src/Core/Extractors/nNASOS.cs

```diff
@@ -0,0 +1,140 @@
+                        else
+                        {
+                            isoDecPath = Directory.GetFiles(cachePath, "*.iso.dec")[0];
+                        }
+                    }
+                }
+                else if (extension.Equals(".iso.dec"))
+                {
+                    Extractor robocopy = new Robocopy();
+                    if (robocopy.Extract(archivePath, cachePath, includeList, excludeList))
+                    {
+                        isoDecPath = Directory.GetFiles(cachePath, "*.iso.dec")[0];
+                    }
+                }
+                else if (extension.Equals(".iso"))
+                {
+                    Extractor robocopy = new Robocopy();
+                    if (robocopy.Extract(archivePath, cachePath, includeList, excludeList))
+                    {
+                        // no need to extract as already in .iso format
+                        return true;
+                    }
+                }
+            }
+
+            // Double-check isoDecPath is set, and not the original archive path, as it will be deleted.
+            if (string.IsNullOrEmpty(isoDecPath) || string.Equals(isoDecPath, archivePath, StringComparison.InvariantCultureIgnoreCase))
+            {
+                Logger.Log($"nNASOS path check failed: {isoDecPath}");
+                Environment.ExitCode = 1;
+                return false;
+            }
+
+            /*
+             * Usage: nNASOS <options> <file1> <file2> <fileN>
+             * Possible options:
+             * -v      Verify
+             * -n      Do not pause at the end of processing
+             * -d      Remove source file at successfull conversion
+             * -p      Pause at error
+             *
+             * It is safe to delete the original .iso.dec since we're working with the
+             * copied/extracted file in the cache.
+             * Also, there is no option to specify an output path, but afaik, it simply extracts
+             * to an .iso in the same folder as input file.
+             */
+            string args = string.Format("-d -n \"{0}\"", isoDecPath);
+            
+            (string stdout, string stderr, int exitCode) = ProcessUtils.RunProcess(executablePath, args);
+
+            if (exitCode != 0)
+            {
+                // errors on stderr stream (at least according to the readme)
+                Logger.Log(string.Format("nNASOS returned exit code {0} with error output:\r\n{1}", exitCode, stderr));
+                Environment.ExitCode = exitCode;
+            }
+
+            return exitCode == 0;
+        }
+
+        public override long GetSize(string archivePath, string fileInArchive = null)
+        {
+            return DiskUtils.GetFileSize(archivePath);
+        }
+
+        public override string[] List(string archivePath)
+        {
+            return string.Format("{0}.iso.dec", Path.GetFileNameWithoutExtension(archivePath)).ToSingleArray();
```

Comment:
This will be the final filename after 'extraction', so should be just .iso
```suggestion
            return string.Format("{0}.iso", Path.GetFileNameWithoutExtension(archivePath)).ToSingleArray();
```